### PR TITLE
Add fieldset component

### PR DESCRIPTION
**Pull request by FlyersPh9**
Adding the [fieldset](https://developer.mozilla.org/en-US/docs/Web/HTML/Element/fieldset) component.

light
dark

### CODE REVIEW

File: backstop/tests/fieldset.html [hunk 1]
```diff
@@ -0,0 +1,194 @@
...
+        <label class="iui-input-container iui-inline iui-disabled">
+          <div class="iui-label">Internal URL</div>
+          <input
+            value="http://127.0.0.1:5500/backstop/minified/fieldset.html"
+            class="iui-input"
+            disabled
+          />
+        </label>
+
+        <label class="iui-toggle-switch">
```
Comment (bentleyvk) [inline backstop/tests/fieldset.html:176]: Accidently saw that when you hover over green area (padding), it shows pointer cursor. 
image
We either need to remove this padding or change it to margin.

File: src/fieldset/fieldset.scss [hunk 2]
```diff
@@ -0,0 +1,36 @@
...
+@import '../inputs/index';
+
+@mixin iui-fieldset {
+  @include iui-reset;
+  @include iui-font-family;
+  padding: $iui-baseline $iui-sm;
+  border-radius: $iui-border-radius;
+  @include themed {
+    border: 1px solid t(iui-color-background-3);
+    color: t(iui-text-color);
```
Comment (bentleyvk) [inline src/fieldset/fieldset.scss:13]: Is this one needed?

Comment (FlyersPh9) [reply src/fieldset/fieldset.scss:13]: It is not, removed.

File: backstop/tests/fieldset.html (hunk 1 above)
Comment (FlyersPh9) [reply backstop/tests/fieldset.html:176]: I tried swapping `padding` with `margin` on toggle switches but it's causing more changes than expected.
Screen Shot 2021-05-13 at 10 10 08 AM
Screen Shot 2021-05-13 at 10 11 01 AM

Comment (bentleyvk) [reply backstop/tests/fieldset.html:176]: Ah yeah, vertical margins collapse. Created an item in our backlog. We will take a look at it later.

**Review by bentleyvk — APPROVED** [review]

**Review by mayank99 — APPROVED** [review]

**Pull request merged**